### PR TITLE
Modify api as figma 

### PR DESCRIPTION
**Pull request by Radhe2911**
Modified the API as per figma.

### CODE REVIEW

File: server/src/controllers/userController.ts [hunk 1]
```diff
@@ -3,19 +3,26 @@ import User, { getUserDetails, updateUser } from "../models/Users.js";
...
+        const { id } = req.user;
 
-        const userDetails = await getUserDetails(userId, next);
+        const userDetails = await getUserDetails(id, next);
         if (!userDetails) {
             return next(new NotFoundError("User not found"))
         }
+
+        const { password, ...safeUserDetails } = userDetails.toObject(); // user details without password
+        console.log("safeUserDetails", safeUserDetails);
```
Comment (tw-dev-alfaiz) [inline server/src/controllers/userController.ts]: remove console.log

File: server/src/validations/validation.ts [hunk 2]
```diff
@@ -126,7 +94,28 @@ export const updateUserValidation = async (req: Request, res: Response, next: Ne
...
+                is: "true",
+                then: joi.string().required()
+                    .label("Password")
+                    .pattern(new RegExp("^(?=.*[a-z])(?=.*[A-Z])(?=.*\\d)(?=.*[@$!%*?&])[A-Za-z\\d@$!%*?&]{8,16}$"))
+                    .messages({
+                        "string.pattern.base": 'Password must be of 8-16 character with uppercase, lowercase, number, and special characters(@$!%*?&)',
+                    }),
+                otherwise: joi.optional()
+            }),
+            // currentPassword: joi.string().required().label("Current Password"),
```
Comment (tw-dev-alfaiz) [inline server/src/validations/validation.ts]: remove commented code if it is not required

File: server/src/controllers/userController.ts [hunk 3]
```diff
@@ -26,27 +33,33 @@ const userDetails = async (req: Request, res: Response, next: NextFunction) => {
...
+        const updatingData = {
+            ...req.body
+        }
+
         const userDetails = await getUserDetails(userId, next);
         if (!userDetails) {
             return next(new NotFoundError("User not found"))
         }
 
+        if (isPasswordChange == "true") {
```
Comment (tw-vineet) [inline server/src/controllers/userController.ts]: why using string 'true' 
use only  this type of condition` if (isPasswordChange)`

File: server/src/controllers/userController.ts [hunk 4]
```diff
@@ -3,19 +3,26 @@ import User, { getUserDetails, updateUser } from "../models/Users.js";
...
         }
+
+        const { password, ...safeUserDetails } = userDetails.toObject(); // user details without password
+        console.log("safeUserDetails", safeUserDetails);
+
         res.status(200).json({
             status: true,
             statusCode: 200,
-            data: userDetails,
+            data: safeUserDetails,
```
Comment (tw-vineet) [inline server/src/controllers/userController.ts]: why use safeUserDetails change this naming convention

File: server/src/controllers/userController.ts [hunk 5]
```diff
@@ -61,14 +74,39 @@ const updateUserDetails = async (req: Request, res: Response, next: NextFunction
...
 
+const userList = async (req: Request, res: Response, next: NextFunction) => {
+    try {
+
+        const allUserList = await User.find({ isAdmin: false });
+        res.status(200).json({
+            status: true,
+            statusCode: 200,
+            data: allUserList,
+            message: "User details",
```
Comment (tw-vineet) [inline server/src/controllers/userController.ts]: all message come from constant file

**Pull request merged**